src/engineer/engineer.py

```diff
@@ -0,0 +1,181 @@
+import numpy as np
+from collections import defaultdict
+import calendar
+from datetime import date
+from pathlib import Path
+import xarray as xr
+
+from typing import cast, Dict, List, Optional, Union, Tuple
+from typing import DefaultDict as DDict
+
+from ..utils import minus_months
+
+
+class Engineer:
+    """The engineer is responsible for taking all the data from the preprocessors,
+    and saving a single training netcdf file to be ingested by machine learning models.
+
+    Training and test sets are defined here, to ensure different machine learning models have
+    access to the same data.
+
+    Attributes:
+    ----------
+    data_folder: Path, default: Path('data')
+        The location of the data folder.
+    """
+
+    def __init__(self, data_folder: Path = Path('data')) -> None:
+
+        self.data_folder = data_folder
+
+        self.interim_folder = data_folder / 'interim'
+        assert self.interim_folder.exists(), \
+            f'{data_folder / "interim"} does not exist. Has the preprocesser been run?'
+
+        self.output_folder = data_folder / 'features'
+        if not self.output_folder.exists():
+            self.output_folder.mkdir()
+
+    def engineer(self, test_year: Union[int, List[int]],
+                 target_variable: str = 'VHI'):
+        """
+        Take all the preprocessed data generated by the preprocessing classes, and turn it
+        into a single training file to be ingested by the machine learning models.
+
+        Arguments
+        ----------
+        test_year: Union[int, List[int]]
+            Data to be used for testing. No data earlier than the earliest test year
+            will be used for training.
+            If a list is passed, a file for each year will be saved.
+        target_variable: str = 'VHI'
+            The variable to be predicted. Only this variable will be saved in the test
+            netcdf files
+        """
+        data = self._make_dataset()
+
+        if type(test_year) is int:
```

Comment:
if test_year is of type `int` then cast test_year to `int`?

Reply:
So if it is an int, I turn it into a list.

The `cast` bit it just to tell mypy that if we're here, we're sure `year` is an `int`, not a `Union[List[int], int]`.